src/topology/mod.rs

```diff
@@ -179,14 +179,8 @@ impl RunningTopology {
             .map(|_| ())
             .map_err(|_| ());
 
-        // TODO: Once all Sources properly look for the ShutdownSignal, remove this in favor of
-        // using 'deadline' instead.
```

Comment:
Is there a way of verifying that this is the case?

Reply:
If there is a source not looking for the `ShutdownSignal`, it should prolong the shutdown for that much. So a test to measure the shutdown length of vector with each/every source and seeing if it's less than the deadline, would do the trick. I'll extend #1089 with it.    

---
Good thing you asked, our stop procedure is still dropping the handles of source tasks, so we aren't actually using `ShutdownSignal` yet. I'll add an issue for it.